sdk/formrecognizer/ai-form-recognizer/src/transforms.ts

```diff
@@ -199,9 +199,9 @@ export function toRecognizedFormArray(
       original.analyzeResult?.documentResults
         ?.filter((d) => !!d.fields)
         ?.map((d) => {
-          if (expectedDocType !== undefined && expectedDocType !== d.docType) {
+          if (expectedDocTypePrefix !== undefined && !d.docType.startsWith(expectedDocTypePrefix)) {
             throw new RangeError(
-              `Expected document type '${expectedDocType}', but found '${d.docType}'.`
+              `Expected document type to start with '${expectedDocTypePrefix}', but found '${d.docType}'.`
```

Comment:
The reason I do it is because custom form and all the prebuilts all use the exact same LRO implementation that is abstracted over a function that returns `RecognizedForm`, and this is basically a sanity check that the right service method ultimately ended up servicing the result. Custom form recognition doesn't validate the doc type at all, but the prebuilts do as a way of making sure that the call was dispatched to the right backend method.

Reply:
got it - thanks!